### DISCUSSION TITLE
How to "upload" code onto ESP32-C3 Super Mini?

### DISCUSSION THREAD

**Discussion by Iced-Coded** [micropython/micropython#16064] [ESP32] [answered]
Hey, so, I have one project which I need to upload onto the ESP32-C3, but PyCharm doesn't support ESP32, as I know.

**Comment by andrewleech** [comment, marked as answer]
I suggest you start with [mpremote](https://docs.micropython.org/en/latest/reference/mpremote.html), in particular the sections about mip and fs / cp.

**Comment by Witawat** [comment]
> Hey, so, I have one project which I need to upload onto the ESP32-C3, but PyCharm doesn't support ESP32, as I know.

Can you write a tutorial on how to upload to esp32 supper mini with pycharm?

I also use pychram but I can't figure it out.

**Reply by Iced-Coded** [reply]
You can't, that's the deal
It just doesn't support PyCharm at all, so, you gotta do it manually.

**Reply by Witawat** [reply]
Can you please share your upload method?

**Reply by Iced-Coded** [reply]
> Can you please share your upload method?

Either use Thonny, or, mpremote

**Comment by sosi-deadeye** [comment]
- Install Python (e.g. 3.13)
- install esptool
  ````
  py -3.13 -m pip install esptool
  ````
- Download the bin-file for your board
- Hold the Boot-Button and connect your ESP32C3
- ```
  py -3.13 -m esptool --port COMxy --baud 460800 erase_flash
  py -3.13 -m esptool --port COMxy --baud 460800 write_flash 0x0 THE-BIN-FILE   
  ```
- Reconnect or reset your board, then you can use it.

More info: https://micropython.org/download/ESP32_GENERIC_C3/

PS: If you mean with uploading Code, not the Firmware, then it's different. Then use mpremote to put files on your controller. An alternative is Thonny, but it has its own quirks.

**Comment by Witawat** [comment]
Now i try PyCharm Upload to Super Mini C3 is work

can code on PyCharm ok
